### PR TITLE
James h65 patch datasheets page

### PR DESCRIPTION
**Pull request by JamesH65**
@lurch Thinking something like this. Not happy yet with it, although cannot place exactly why.

### CODE REVIEW

File: hardware/raspberrypi/datasheets.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Datasheets
+
+### SoC's
+
+The following datasheets are available for the SoC (System On Chip - the main processing chip) devices used on the Raspberry Pi platform.
+
+[BCM2711](bcm2711/rpi_DATA_2711_1p0.pdf) - The SoC used on the Raspberry Pi 4.
+
+[BCM2835](bcm2735/BCM2835-ARM-Peripherals.pdf) - The SoC used on the Raspberry Pi 0 and 1. This datasheet is also applicable to the Raspberry Pi 2 and 3, with the BCM2836 and BC2837 SoC's respectively, as the Videocore GPU is the same.
```
Comment (lurch) [inline hardware/raspberrypi/datasheets.md:9]: BCM2835 was released before BCM2711, so I'd list it first. :man_shrugging: 
Also, I'm assuming the `bcm2735` in the URL is wrong? :laughing:

File: hardware/raspberrypi/README.md [hunk 2]
```diff
@@ -4,6 +4,8 @@ The hardware in the Raspberry Pi
 
 - [Schematics](schematics/README.md)
     - Schematics for the Raspberry Pi
+- [Datasheets](datasheets.md)
+    - Currently available datasheets for the Raspberry Pi range
```
Comment (lurch) [inline hardware/raspberrypi/README.md:8]: The schematics link doesn't say "Currently available schematics for the Raspberry Pi range", so maybe the text here should just say "Datasheets for the Raspberry Pi" ?

**Comment by lurch** [comment]
There's also the "ARM-local peripherals" document on the BCM2836 page - is that worth linking here?

File: hardware/raspberrypi/datasheets.md (hunk 1 above)
Comment (JamesH65) [reply hardware/raspberrypi/datasheets.md:9]: Weird, I thought I checked all the links...

File: hardware/raspberrypi/README.md (hunk 2 above)
Comment (JamesH65) [reply hardware/raspberrypi/README.md:8]: Just trying to add a bit of variety!

**Comment by JamesH65** [comment]
I wasn't sure what to do about that arm stuff. But having taken a longer look, I think we should have it in there. I wonder how much of it is relevant to the generic Quad core arm devices - ie A7->A53->A72.

File: hardware/raspberrypi/datasheets.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+## Datasheets
+
+### SoC's
+
+The following datasheets are available for the SoC (System On Chip - the main processing chip) devices used on the Raspberry Pi platform.
+
+[BCM2835](bcm2835/BCM2835-ARM-Peripherals.pdf) - The SoC used on the Raspberry Pi 0 and 1. This datasheet is also applicable to the Raspberry Pi 2 and 3, with the BCM2836 and BC2837 SoC's respectively, as the Videocore GPU is the same.
```
Comment (lurch) [inline hardware/raspberrypi/datasheets.md:7]: This feels a little bit clumsy? Perhaps:
The SoC used on the Raspberry Pi 0 and 1. This datasheet is also applicable to the BCM2836 and BC2837 SoCs (used on the Pi2 and Pi3 respectively), as they use the same VideoCore IV GPU.

:question:

**Comment by lurch** [comment]
> I wonder how much of it is relevant to the generic Quad core arm devices - ie A7->A53->A72.

AFAIK it applies to both the BCM2836 and BCM2837, but **doesn't** apply to BCM2711.

File: hardware/raspberrypi/datasheets.md (hunk 3 above)
Comment (lurch) [reply hardware/raspberrypi/datasheets.md:7]: ...or perhaps "Large parts of this datasheet are also applicable to the BCM2836 and BC2837 SoCs ..." ?

Comment (JamesH65) [reply hardware/raspberrypi/datasheets.md:7]: Yes, much better. Ta.

**Comment by JamesH65** [comment]
> > I wonder how much of it is relevant to the generic Quad core arm devices - ie A7->A53->A72.
> 
> AFAIK it applies to both the BCM2836 and BCM2837, but **doesn't** apply to BCM2711.

That would be my guess, but would need to confirm.

File: hardware/raspberrypi/datasheets.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+## Datasheets
+
+### SoC's
```
Comment (lurch) [inline hardware/raspberrypi/datasheets.md:3]: Greengrocer's apostrophe? :wink:

**Comment by pelwell** [comment]
Yes, it's applicable to 2837. If you look at bcm2837.dtsi in the kernel
you'll see that it makes use of the bcm2836-l1-intc and bcm2836-smp drivers,
which make use of the hardware described in that PDF.

**Comment by JamesH65** [comment]
OK, will update the links and text appropriately. Thanks all.

File: hardware/raspberrypi/bcm2837/README.md [hunk 5]
```diff
@@ -4,4 +4,8 @@ This is the Broadcom chip used in the Raspberry Pi 3, and in later models of the
 
 The ARM cores run at 1.2GHz, making the device about 50% faster than the Raspberry Pi 2. The VideoCore IV runs at 400MHz.
 
+Please refer to the following BCM2836 document for details on the ARM peripherals specification which is unchanged.
```
Comment (pelwell) [inline hardware/raspberrypi/bcm2837/README.md:7]: "...which also applies to the BCM2837." is less ambiguous.

Comment (JamesH65) [reply hardware/raspberrypi/bcm2837/README.md]: Done

**Comment by JamesH65** [comment]
Thanks all.

**Pull request merged**